### PR TITLE
Docs enhancements

### PR DESCRIPTION
**Pull request by echarles**
A few proposals to make the docs more readable and fix minor issues (links...):

+  fix broken download link and use latest release
+ add missing yml extension to docker-compose file
+ dedicated section on orechtestrators in toc
+ user kern-... as file name to describe kernels running on resource managers / orchestrators

### CODE REVIEW

File: docs/source/getting-started-kernels.md [hunk 1]
```diff
@@ -7,12 +7,22 @@ The following kernels have been tested with the Jupyter Enterprise Gateway:
...
+We provide sample kernel configuration and launcher tar files as part of [each release](https://github.com/jupyter/enterprise_gateway/releases) (e.g. [jupyter_enterprise_gateway_kernelspecs-2.0.0.dev2.tar.gz](https://github.com/jupyter/enterprise_gateway/releases/download/v2.0.0rc1/jupyter_enterprise_gateway_kernelspecs-2.0.0rc1.tar.gz)) that can be extracted and modified to fit your configuration.
 
 Please find below more details on the specific kernels supported by Enterprise Gateway.  In each of the sections we set the `KERNELS_FOLDER` to `/usr/local/share/jupyter/kernels` since that's one of the default locations searched by the Jupyter framework.  Co-locating kernelspecs hierarchies in the same parent folder is recommended, although not required.
 
 For information about how to build your own kernel-based docker image for use by Enterprise Gateway see [Custom kernel images](docker.html#custom-kernel-images).
 
+You will find in this documentation details on how to install the kernels on various orchestrators:
+
++ [Hadoop Yarm client mode](kern-yarn-client-mode.md.html).
++ [Hadoop Yarm cluster mode](kern-yarn-cluster-mode.html).
```
Comment (kevin-bates) [inline docs/source/getting-started-kernels.md]: Please fix the typos of these Yarn entries and let's drop the period at the end of each of these bullets.

File: docs/source/index.rst [hunk 2]
```diff
@@ -26,15 +26,20 @@ distributes kernels across the compute cluster, dramatically increasing the numb
...
    getting-started-kernels
    getting-started-security
    getting-started-other-features
 
+.. toctree::
+   :maxdepth: 2
+   :caption: Resource Managers
+
+   kern-yarn-cluster-mode
+   kern-yarn-client-mode.md
```
Comment (kevin-bates) [inline docs/source/index.rst]: I don't think we want the `.md` here.

File: docs/source/kern-docker.md [hunk 3]
```diff
@@ -13,7 +14,7 @@ The current deployment uses a compose stack definition, [docker-compose.yml](htt
 To deploy the stack to a swarm cluster from a manager node, use:
 
 ```bash
-docker stack deploy -c docker-compose enterprise-gateway
+docker stack deploy -c docker-compose.yml enterprise-gateway
```
Comment (kevin-bates) [inline docs/source/kern-docker.md]: Nice catch

File: docs/source/system-architecture.md [hunk 4]
```diff
@@ -256,7 +256,7 @@ Like the yarn endpoint parameter the `remote_hosts` parameter can be specified w
 process proxy configuration to override the global value - enabling finer-grained kernel distributions.
 
 See 
-[Enabling YARN Client Mode or Spark Standalone Support](getting-started-client-mode.html#enabling-yarn-client-mode-or-spark-standalone-support) for details.
+[Enabling YARN Client Mode or Spark Standalone Support](kern-yarn-client-mode.md.html#enabling-yarn-client-mode-or-spark-standalone-support) for details.
```
Comment (kevin-bates) [inline docs/source/system-architecture.md]: `.md` should be removed.

File: docs/source/getting-started-kernels.md [hunk 5]
```diff
@@ -7,12 +7,22 @@ The following kernels have been tested with the Jupyter Enterprise Gateway:
...
++ [Kubernetes](kern-kubernetes.html).
++ [Docker](kern-docker.html).
++ [IBM Conductor](kern-conductor.html).
+
+For the impatients, we give here after basic information for Yarn.
```
Comment (kevin-bates) [inline docs/source/getting-started-kernels.md]: I don't understand this line.  What are you trying to say?

File: docs/source/getting-started-kernels.md (hunk 1 above)
Comment (echarles) [reply docs/source/getting-started-kernels.md]: fixed

File: docs/source/kern-docker.md (hunk 3 above)
Comment (echarles) [reply docs/source/kern-docker.md]: fixed

**Comment by echarles** [comment]
Thx @kevin-bates , I have pushed a new commit fixing your comments.

Regarding #653 I already shortened the names (forgot to list in the changes): e.g. `Enabling Kubernetes Support` is now `Kernels on Kubernetes`

I have removed the line "For the impatients...": The reason was that at the `getting-started-kernels.html` there is a bried description of how to setup 3 kernel languages: scala on yarn-cluster, python on yarn-client, R on yarn-client. After, for each of the resource manager, we start again describing the setup in dedicated pages without details on the languages. A radical mean to fix this ambiguity/partial information would be to remote the section on the getting started page and bring details for languages on each of the resource manager pages. This may duplicate sections, but all language do not apply to all resource managers the same way. WDYT.

PS: is travis stable? it failed with timeout when I opened the PR (unrelated to my changes I think). Let's how it is based on my new push.

**Comment by echarles** [comment]
One more open point is the single page for `yarn-client` and `spark-cluster`: The beginning of that page is common to both but I would better see 2 separate pages for them. WDYT

**Comment by echarles** [comment]
For travis, the first time `python 3.6` profile has succeeded. This time both profiles (3.6 and 3.5) are failing with IMHO reasons unrelated to this PR.

**Comment by kevin-bates** [comment]
Hi Eric - yeah, the travis stuff is fairly flakey wrt to what we're trying to do.  We've found 3.5 to be much less stable, but occasionally both will fail.  I believe there's a timing issue with port cleanup (at the os level) that then interferes with subsequent tests, but haven't really drilled into things.  I also see travis timeout prior to the launch of docker image used for integration tests - with no expected output prior to the launch.  Its very frustrating and has nothing to do with your changes.

**Pull request merged**